x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/QueryRuleCriteria.java

```diff
@@ -167,20 +192,57 @@ public boolean equals(Object o) {
         QueryRuleCriteria that = (QueryRuleCriteria) o;
         return criteriaType == that.criteriaType
             && Objects.equals(criteriaMetadata, that.criteriaMetadata)
-            && Objects.equals(criteriaValue, that.criteriaValue);
+            && Objects.equals(criteriaValues, that.criteriaValues);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(criteriaType, criteriaMetadata, criteriaValue);
+        return Objects.hash(criteriaType, criteriaMetadata, criteriaValues);
     }
 
     @Override
     public String toString() {
         return Strings.toString(this);
     }
 
-    public boolean isMatch(String matchString) {
-        return criteriaType == CriteriaType.EXACT && criteriaValue.equals(matchString);
+    public boolean isMatch(Object matchValue, CriteriaType matchType) {
+        if (criteriaType == GLOBAL) {
+            return true;
+        }
+
+        final LevenshteinDistance ld = new LevenshteinDistance();
+        final String matchString = matchValue.toString();
+        boolean matchFound = false;
+
+        for (Object criteriaValue : criteriaValues) {
+            final String criteriaValueString = criteriaValue != null ? criteriaValue.toString() : null;
+            if (criteriaValueString != null && matchValue instanceof String) {
+                matchFound |= switch (matchType) {
```

Comment:
I find confusing to use the non-short-circuit version of the or operator. We also want to break out of the loop as soon as we find a match, right?

Reply:
We don't need break statements with the arrow syntax sugar.

Reply:
Got it, good catch, thanks!